src/libsyntax/errors/emitter.rs

```diff
@@ -24,27 +25,15 @@ use std::rc::Rc;
 use term;
 
 pub trait Emitter {
-    fn emit(&mut self, span: Option<&MultiSpan>, msg: &str, code: Option<&str>, lvl: Level);
-    fn custom_emit(&mut self, sp: &RenderSpan, msg: &str, lvl: Level);
+    fn emit(&mut self, span: &MultiSpan, msg: &str, code: Option<&str>, lvl: Level);
 
     /// Emit a structured diagnostic.
-    fn emit_struct(&mut self, db: &DiagnosticBuilder) {
-        self.emit(db.span.as_ref(), &db.message, db.code.as_ref().map(|s| &**s), db.level);
-        for child in &db.children {
-            match child.render_span {
-                Some(ref sp) => self.custom_emit(sp, &child.message, child.level),
-                None => self.emit(child.span.as_ref(), &child.message, None, child.level),
-            }
-        }
-    }
+    fn emit_struct(&mut self, db: &DiagnosticBuilder);
```

Comment:
Actually, why is `custom_emit` gone? It seems like if we're keeping `RenderSpan` around (which I never really liked), it makes sense to have API to emit it - I can't imagine implementing an Emitter that doesn't.